api/src/pcapi/core/subscription/api.py

```diff
@@ -88,6 +88,18 @@ def activate_beneficiary(user: users_models.User) -> users_models.User:
 
 
 def has_completed_profile(user: users_models.User) -> bool:
+    # if the user has a pending DMS subscription, we consider the profile as potentially completed
+    # (it will be completed when the subscription is activated)
+    user_pending_dms_fraud_checks = [
+        fraud_check
+        for fraud_check in user.beneficiaryFraudChecks
+        if fraud_check.type == fraud_models.FraudCheckType.DMS
+        and fraud_check.status in (fraud_models.FraudCheckStatus.PENDING, fraud_models.FraudCheckStatus.STARTED)
+        and fraud_check.resultContent["city"] is not None  # TODO: remove when all DMS applications ask for city
+    ]
+    if user_pending_dms_fraud_checks:
+        return True
```

Comment:
optim de perf : on peut faire cette condition en fallback si `all(elem is not None for elem in mandatory_fields)` est faux